File: Go/Algorithms/combSort.go
package main

func CombSort(arr []int) {
	n := len(arr)
	gap := n
	swapped := true

	for gap > 1 || swapped {
			gap = int(float64(gap) / 1.3)
			if gap < 1 {
					gap = 1
			}

			swapped = false
			for i := 0; i < n-gap; i++ {
					if arr[i] > arr[i+gap] {
							temp := arr[i]
							arr[i] = arr[i+gap]
							arr[i+gap] = temp
							swapped = true
					}
			}
	}
}

File: Nim/Algorithms/comb_sort.nim
proc comb_sort(arr: var seq[int]): seq[int] = 
    var 
        n: int = arr.len
        gap: int = n
        swapped: bool = true

    while gap > 1 or swapped:
        gap = (gap.float / 1.3).int

        if gap < 1:
            gap = 1
        
        swapped = false

        for i in 0..<n-gap:
            if arr[i] > arr[i+gap]:
                var temp: int = arr[i]
                arr[i] = arr[i+gap]
                arr[i+gap] = temp
                swapped = true
    
    return arr